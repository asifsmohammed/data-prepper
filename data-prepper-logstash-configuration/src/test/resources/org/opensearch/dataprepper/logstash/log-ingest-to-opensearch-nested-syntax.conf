input {
    http {
        port => 3000
    }
}
filter {
    grok {
        match => {"[data][log1]" => "%{COMBINEDAPACHELOG}"}
        match => ["[data][log2]", "%{COMBINEDAPACHELOG}"]
        match => {
                  "message" => [
                    "Duration: %{NUMBER:duration}",
                    "Speed: %{NUMBER:speed}"
                  ]
                }
        break_on_match => false
    }
    drop { }
    kv {
        target => "test"
    }
    mutate {
        add_field => { "message1" => 3 "message2" => 4.2 "message3" => "test3" "message4" => true }
        rename => { "messagea" => "newMessage" "messageb" => "newMessage2" }
        copy => { "messagec" => "messaged" }
        remove_field => ["test", "test2"]
        remove_field => ["test3", "test4"]
    }
}
output {
    elasticsearch {
        hosts => ["https://localhost:19000"]
        user => myuser
        password => mypassword
        index => "simple-pipeline"
    }
}