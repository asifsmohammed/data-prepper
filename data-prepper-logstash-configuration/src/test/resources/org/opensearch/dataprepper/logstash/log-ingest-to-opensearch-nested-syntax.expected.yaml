logstash-converted-pipeline:
  source:
    http:
      max_connection_count: 500
      request_timeout: 10000
      port: 3000
  processor:
    - grok:
        keys_to_overwrite:
          - "/data/timestamp"
          - "/data/verb"
        break_on_match: false
        match:
          /data/log1:
            - "%{COMBINEDAPACHELOG}"
          /data/log2:
            - "%{COMBINEDAPACHELOG}"
    - date:
        from_time_received: false
        destination_timezone: "UTC"
        destination: "/data/@timestamp"
        match:
          - key: "/data/timestamp"
            patterns:
              - "yyyy-MM-dd"
    - key_value:
        source: "/data/message"
        destination: "/data/test"
    - rename_keys:
        entries:
          - from_key: "/messages/messagea"
            to_key: "/newMessage"
          - from_key: "/messages/messageb"
            to_key: "/newMessage2"
    - copy_values:
        entries:
          - from_key: "/messages/messagec"
            to_key: "/messaged"
    - add_entries:
        entries:
          - key: "/messages/message1"
            value: 3
          - key: "/messages/message2"
            value: 4.2
          - key: "/messages/message3"
            value: "test3"
          - key: "/messages/message4"
            value: true
    - delete_entries:
        with_keys:
          - "/messages/test"
          - "/test2"
          - "/messages/test3"
          - "/test4"
  sink:
    - opensearch:
        hosts:
          - "https://localhost:19000"
        username: "myuser"
        password: "mypassword"
        index: "simple-pipeline"
